axelrod/strategy_transformers.py

```diff
@@ -424,15 +461,33 @@ def joss_ann_wrapper(player, opponent, proposed_action, probability):
     if sum(probability) > 1:
         probability = tuple([i / sum(probability) for i in probability])
 
+    if 1 not in probability or max(probability) == 0:
+        player.classifier["stochastric"] = True
+
     remaining_probability = max(0, 1 - probability[0] - probability[1])
     probability += (remaining_probability,)
     options = [C, D, proposed_action]
     action = choice(options, p=probability)
     return action
 
+def jossann_reclassifier(original_classifier, probability):
+    """
+    Reclassify: note that if probabilities are (0, 1) or (1, 0) then we override
+    the original classifier.
+    """
+    if sum(probability) > 1:
+        probability = tuple([i / sum(probability) for i in probability])
+
+    if probability[0] in [0, 1]:
```

Comment:
I don't think so because of the normalisation step that happens before. I'm fixing some remaining failures and will take a look at the (0, 0) case! :+1: